wolfcrypt/src/asn.c

```diff
@@ -9956,29 +9956,24 @@ int wc_EccPrivateKeyDecode(const byte* input, word32* inOutIdx, ecc_key* key,
     XMEMCPY(priv, &input[*inOutIdx], privSz);
     *inOutIdx += length;
 
-    if ((*inOutIdx + 1) > inSz) {
-    #ifdef WOLFSSL_SMALL_STACK
-        XFREE(priv, NULL, DYNAMIC_TYPE_TMP_BUFFER);
-        XFREE(pub, NULL, DYNAMIC_TYPE_TMP_BUFFER);
-    #endif
-        return BUFFER_E;
-    }
-
-    /* prefix 0, may have */
-    b = input[*inOutIdx];
-    if (b == ECC_PREFIX_0) {
-        *inOutIdx += 1;
+    if (ret == 0 && (*inOutIdx + 1) < inSz) {
```

Comment:
The context specific 0 and 1 parts of the ASN.1 are optional. The change makes sure that either or both can be missing. Only works if the curve is the default curve.